src/ehrdata/core/ehrdata.py

```diff
@@ -1,240 +1,129 @@
-
-    @property
-    def obsp(self):
-        """Field from AnnData."""
-        return self._adata.obsp
-
-    @obsp.setter
-    def obsp(self, input):
-        self._adata.obsp = input
-
-    @property
-    def varp(self):
-        """Field from AnnData."""
-        return self._adata.varp
-
-    @varp.setter
-    def varp(self, input):
-        self._adata.varp = input
-
-    @property
-    def var_names(self):
-        """Field from AnnData."""
-        return self._adata.var_names
-
-    @var_names.setter
-    def var_names(self, input):
-        self._adata.var_names = input
-
-    @property
-    def obs_names(self):
-        """Field from AnnData."""
-        return self._adata.obs_names
-
-    @obs_names.setter
-    def obs_names(self, input):
-        self._adata.obs_names = input
-
-    @property
-    def uns(self):
-        """Field from AnnData."""
-        return self._adata.uns
-
-    @uns.setter
-    def uns(self, input):
-        self._adata.uns = input
-
-    @property
-    def layers(self):
-        """Field from AnnData."""
-        # TODO: dont allow writing a specific key to layers..?
-        return self._adata.layers
-
-    @layers.setter
-    def layers(self, key, input):
-        # TODO: if 3D: need to match time dimension?
-        self._adata.layers[key] = input
+            raise ValueError("t must be a pandas.DataFrame")
+
+    @classmethod
+    def from_adata(cls, adata: AnnData, *, t: pd.DataFrame | None = None) -> EHRData:
+        """Create an EHRData object from an AnnData object."""
+        instance = cls(shape=adata.shape, t=t)
+        if adata.is_view:
+            instance._init_as_view(adata, slice(None), slice(None))
+        else:
+            instance._init_as_actual(
+                X=adata.X,
+                obs=adata.obs,
+                var=adata.var,
+                uns=adata.uns,
+                obsm=adata.obsm,
+                varm=adata.varm,
+                obsp=adata.obsp,
+                varp=adata.varp,
+                raw=adata.raw,
+                layers=adata.layers,
+                shape=adata.shape if adata.X is None else None,
+                filename=adata.filename,
+                filemode=adata.file._filemode,
+            )
+        return instance
 
     @property
-    def r(self):
+    def r(self) -> np.ndarray | None:
         """3-Dimensional tensor, aligned with obs along first axis, var along second axis, and allowing a 3rd axis."""
-        return self._adata.layers.get(R_LAYER_KEY)
+        return self.layers.get(R_LAYER_KEY)
 
     @r.setter
-    def r(self, input):
-        self._adata.layers[R_LAYER_KEY] = input
+    def r(self, input: np.ndarray | None) -> None:
+        if input is None:
+            del self.r
+        else:
+            self.layers[R_LAYER_KEY] = input
+
+    @r.deleter
+    def r(self) -> None:
+        del self.layers[R_LAYER_KEY]
 
     @property
-    def t(self):
+    def t(self) -> pd.DataFrame | None:
```

Comment:
this property isn’t really necessary right now, maybe just replace `_t: pd.DataFrame` above with `t: pd.DataFrame`.

Reply:
Yeah hahah true I think this was one of the first attributes I've made and considered so ancient I didnt dare to switch during hackathon time.
If you see it too, I do now